### DISCUSSION TITLE
How is the recording output date time generated?

### DISCUSSION THREAD

**Discussion by NikeNut** [danielfernau/unifi-protect-video-downloader#282] [Q&A] [answered]
First of all, thank you for building this!

I would like to have the output subfolders and file name be my local time, but it is be written out in UTC time. Through the Unifi OS GUI, I have the time zone set to my local zone. After ssh-ing and running `timedatectl`, the output is the following:

```bash
               Local time: Sun 2023-02-19 20:30:09 EST
           Universal time: Mon 2023-02-20 01:30:09 UTC
                 RTC time: Mon 2023-02-20 01:30:10
                Time zone: EST5EDT (EST, -0500)
System clock synchronized: yes
              NTP service: active
          RTC in local TZ: no
```

```bash
root@CloutKey:~# hwclock
2023-02-19 20:36:24.607507-05:00
```

Do I need to change the RTC time?

Any suggestions?

**Comment by danielfernau** [comment]
Hi @NikeNut 

That's a really good question, I had someone in the past ask if they could get paths and file names in UTC, so I implemented the `--use-utc-filenames` option... Unless that's added to the command line, it uses the default format for the environment. It does not care about the local time of the CloudKey, only that of the Docker host machine (unless, of course, you're using it on the CloudKey directly).

Could you please post the command line you're using (credentials/IPs removed) here along with some info about where you run the tool on?

Thanks!

**Reply by NikeNut** [reply]
I am running Docker on a M2 (Apple Silicon), MacOS 13.2 (Ventura).

From within a shell script, I have these commands running automatically via Launchd:
```shell
start_date=$(date -v-30M +"%Y-%m-%d %H:%M:%S%z")
end_date=$(date +"%Y-%m-%d %H:%M:%S%z")
/usr/local/bin/docker run --volume /path/on/host/machine:/downloads unifitoolbox/protect-archiver events  --start "$start_date" --end "$end_date" --ignore-failed-downloads /downloads
```

Does the `TZ` environment variable need to be set within the Docker image configuration?

**Comment by NikeNut** [comment, marked as answer]
I solved this by passing the desired timezone into the `TZ` environment variable.

```shell
start_date=$(date -v-30M +"%Y-%m-%d %H:%M:%S%z")
end_date=$(date +"%Y-%m-%d %H:%M:%S%z")
/usr/local/bin/docker run -e TZ=America/New_York --volume /path/on/host/machine:/downloads unifitoolbox/protect-archiver events  --start "$start_date" --end "$end_date" --ignore-failed-downloads /downloads
```

File output is now in my local time instead of UTC.

Full list of [tz database timezones](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones) if anyone is interested.

**Comment by danielfernau** [comment]
Hi there and sorry for the late reply! Thanks for reporting this issue as well as providing a solution. After a bit of testing on my end I've been able to confirm the problem and will look into it as soon as I can. A fix for this should be relatively easy to implement, as far as I can see. Let me convert this conversation into an issue, that way you'll be notified again once it's been fixed in the code.

Again, thanks for your contribution and for using the tool!
